### PR TITLE
adds token refresh to strata calls

### PR DESCRIPTION
**Pull request by praniredhat**
Please review @vrathee

### CODE REVIEW

File: strata.js [hunk 1]
```diff
@@ -885,7 +895,9 @@
                 onFailure('Error ' + xhr.status + ' ' + xhr.statusText, xhr, reponse, status);
             }
         });
-        $.ajax(fetchCaseComments);
+
+        isTokenExpired() && await window.sessionjs.updateToken(true);
```
Comment (vrathee) [inline strata.js]: this can be removed now..

File: strata.js [hunk 2]
```diff
@@ -919,7 +931,8 @@
                 onFailure('Error ' + xhr.status + ' ' + xhr.statusText, xhr, reponse, status);
             }
         });
-        $.ajax(fetchCaseExternalUpdates);
+
+        isTokenExpired() && await window.sessionjs.updateToken(true);
```
Comment (vrathee) [inline strata.js]: we can remove this now

File: strata.js [hunk 3]
```diff
@@ -95,9 +95,34 @@
...
+    }
+    // If the token is expiring within 30 seconds, go ahead and refresh it.  Using 30 seconds considering jwt.js checks if
+    // the token needs to be refreshed every 58 seconds with a TTE of 90 seconds.  So 30 seconds guarantees that
+    // we are at the boundary of what jwt.js does without overlapping a great deal
+    function isTokenExpired() {
+        var _isTokenExpired = getIsTokenExpired();
+        return _isTokenExpired(30);
+    }
+
+    function callAjaxAndHandleJWT(ajaxObj) {
```
Comment (vrathee) [inline strata.js]: I think we should rename `ajaxObj` to `ajaxParams`

Comment (praniredhat) [reply strata.js]: done

File: strata.js (hunk 2 above)
Comment (praniredhat) [reply strata.js]: oops!!! done

File: strata.js (hunk 1 above)
Comment (praniredhat) [reply strata.js]: oops!! done

**Comment by vrathee** [comment]
Reviewed

**Pull request merged**